### ISSUE TITLE
Used cached edm::View<> when doing a edm::Event::getBy*

### ISSUE THREAD

**Issue by Dr15Jones** [cms-sw/cmssw#12609]
Instead of creating a new ```edm::View<T>``` each time we call ```edm::Event::getBy*<View<T>>``` we should look to see if we've already cached that item and return it. This would have solved the problem seen in this hypernews threads.
https://hypernews.cern.ch/HyperNews/CMS/get/edmFramework/3603.html

**Comment by Dr15Jones** [comment]
One potentially solution would be to move the handling of ```edm::View``` into the ```edm::Principal``` and then the ```getBy*``` do not have to be specialized. Instead, we would use the consumes information to add new ```ProductHolder```s to the ```edm::Principal``` and these holders would handle getting the necessary data from the principal, creating the ```edm::View``` and then the regular caching would handle the lifetime of the ```edm::View```. One tricky part is we'd have an implicit dependency relationship between the holder and the original product. However, since the dependency relationship is strickly based on the module label and that would remain the same there shouldn't be any problem.

**Comment by Dr15Jones** [comment]
assign core

**Comment by smuzaffar** [comment]
@Dr15Jones , @makortel is it still a needed feature?

**Comment by makortel** [comment]
Yes, AFAICT it has not been addressed yet (and based on description so far it would be useful).

**Comment by makortel** [comment]
I prefer to keep it open
